app/routes/announcements/components/AnnouncementsCreate.js

```diff
@@ -0,0 +1,136 @@
+// @flow
+
+import React from 'react';
+import styles from './AnnouncementsList.css';
+import Flex from 'app/components/Layout/Flex';
+import { Form, SelectInput, TextArea } from 'app/components/Form';
+import { reduxForm, Field, reset } from 'redux-form';
+import Button from 'app/components/Button';
+import { ContentMain } from 'app/components/Content';
+
+type Props = {
+  createAnnouncement: (...any) => void,
```

Comment:
Would be best if you made an `AnnouncementEntity` that determines what fields an announcement should have. Probably need one for displaying them and one to determine what fields are sent when creating one, like `CreateAnnouncementEntity` or some such. See reducers for examples. Also, this function probably returns `Promise<*>`, not void.

Reply:
Create a model for it here: https://github.com/webkom/lego-webapp/blob/master/app/models.js

Reply:
I guess I need to move all of my Entities there.. 😅